sockeye/model.py

```diff
@@ -317,6 +317,17 @@ def load_parameters(self,
                                 cast_dtype=cast_dtype, dtype_source=dtype_source)
         logger.info('Loaded params from "%s" to "%s"', filename, mx.cpu() if ctx is None else ctx)
 
+    def set_parameters(self, new_params: dict):
+        """
+        Update model params with new values from a dictionary.
```

Comment:
What is the expected behavior if the shape/context of the new parameter does not match the existing parameter? Can you document this and validate via some unit tests?

Reply:
The parameters are updated on all contexts of the model, regardless of the contexts of the new parameters. I've updated the docstring and added a relevant unit test.

If the shape doesn't match, this will fail, so I added a check for the shapes as well.